trie/ctrie/ctrie.go

```diff
@@ -340,20 +339,80 @@ func (c *Ctrie) ReadOnlySnapshot() *Ctrie {
 	}
 }
 
+// Clear removes all keys from the Ctrie.
+func (c *Ctrie) Clear() {
+	for {
+		root := c.readRoot()
+		gen := &generation{}
+		newRoot := &iNode{
+			main: &mainNode{cNode: &cNode{array: make([]branch, 0), gen: gen}},
+			gen:  gen,
+		}
+		if c.rdcssRoot(root, gcasRead(root, c), newRoot) {
+			return
+		}
+	}
+}
+
+// Iterator returns a channel which yields the Entries of the Ctrie.
+func (c *Ctrie) Iterator() <-chan *Entry {
+	ch := make(chan *Entry)
+	snapshot := c.ReadOnlySnapshot()
+	go func() {
+		traverse(snapshot.root, ch)
+		close(ch)
+	}()
+	return ch
+}
+
+// Size returns the number of keys in the Ctrie.
+func (c *Ctrie) Size() uint {
+	// TODO: The size operation can be optimized further by caching the size
+	// information in main nodes of a read-only Ctrie – this reduces the
+	// amortized complexity of the size operation to O(1) because the size
+	// computation is amortized across the update operations that occurred
+	// since the last snapshot.
+	size := uint(0)
+	for _ = range c.Iterator() {
```

Comment:
Is that compatible with go 1.3? Are we still trying to ensure compatibility with 1.3?